hibernate-core/src/test/java/org/hibernate/orm/test/proxy/concrete/ConcreteProxyWithSealedClassesTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.orm.test.proxy.concrete;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import org.hibernate.annotations.ConcreteProxy;
+import org.hibernate.testing.orm.junit.EntityManagerFactoryScope;
+import org.hibernate.testing.orm.junit.JiraKey;
+import org.hibernate.testing.orm.junit.Jpa;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests the compatibility of Hibernate {@code @ConcreteProxy} with Java {@code sealed}
+ * inheritance hierarchies. Proxy generation may fail during {@code SessionFactory}
+ * bootstrap if the abstract sealed root cannot be subclassed at runtime. These tests
+ * document the limitation and assert the expected failure behavior.
+ *
+ * @author Vincent Bouthinon
+ */
+@Jpa(annotatedClasses = {ConcreteProxyWithSealedClassesTest.Actor.class, ConcreteProxyWithSealedClassesTest.Postman.class})
+@JiraKey("HHH-19899")
+class ConcreteProxyWithSealedClassesTest {
+
+	@Test
+	void testConcreteProxyWithSealedClassesTest(EntityManagerFactoryScope scope) {
+		var id = scope.fromTransaction( entityManager -> {
+			Actor actor = new Postman();
+			entityManager.persist( actor );
+			return actor.id;
+		} );
+		scope.inTransaction( entityManager -> {
+			Actor actor = entityManager.getReference( Actor.class, id );
+			assertThat( actor ).isInstanceOf( Postman.class );
```

Comment:
Would be nice to also have a test case for a `@ManyToOne(fetch=LAZY) Actor actor;` property too, not that I'm worried it won't work, just to be more safe wrt regressions.

Reply:
Good idea. Done.